test/mpi/comm/cmsplit_type.c

```diff
@@ -16,7 +16,7 @@
 static const char *split_topo[] = {
     "machine", "socket", "package", "numanode", "core", "hwthread", "pu", "l1cache",
     "l1ucache", "l1dcache", "l1icache", "l2cache", "l2ucache",
-    "l2dcache", "l2icache", "l3cache", "l3ucache", "l3dcache", "l3icache", NULL
+    "l2dcache", "l2icache", "l3cache", "l3ucache", "l3dcache", "l3icache", "pci:x", NULL
```

Comment:
How will this work?  Shouldn't passing `pci:x` result in a `NULL` communicator since there's no such device?

Reply:
The user asked us to split out the processes that are closest to the "x"th IB device.  What is your proposal for returning in that case?

Reply:
I thought that an invalid id (x in this case) would return a communicator for all processes on the node.

Reply:
Well, it's not defined anywhere and you get to define it the way you want to.  Is that what you think the user would expect?

Reply:
NULL communicator would be the right thing to do, I'll change that.

Reply:
I just noticed that we previously defined the behavior to return a node-wide communicator when an empty info value or garbage info value is passed with shmem_topo info key. Do you think we ought to change that too?

Reply:
Hmm.  That's a good point.  Whichever way we pick, we should be consistent.  How about this:

 - If we do not understand the info value (e.g., `shmem_topo=foobar`), we ignore it (the same as what we are doing today).

 - If we understand the info value, but the corresponding device is not present on the system (e.g., `shmem_topo=ib224`), we return `NULL`.

Is that reasonable?  I'm open to alternate definitions if you disagree.

Reply:
Agreed, I will update the PR as per this definition.